src/client/client.ts

```diff
@@ -197,9 +166,9 @@ async function updateModule(
   } else {
     // dep update
     for (const { deps } of mod.callbacks) {
-      if (Array.isArray(deps)) {
+      if (Array.isArray(deps) && deps.includes(changedPath)) {
         deps.forEach((dep) => modulesToUpdate.add(dep))
-      } else {
+      } else if (deps === changedPath) {
```

Comment:
Removal of the check is intentional. This should be reverted.

Reply:
Also let's avoid mixing unrelated changes in the same PR.

Reply:
Copy that.
But I change it becasue any change will bubbles up to the boundary  when i use hmr api for refactor vue file hmr.That will cause hmr don‘t minimize update for vue file.And also maybe the pr is unnessary.

Reply:
I revert this hmr change already.we can just do it with current hmr api.